src/components/Card/CardList.tsx

```diff
@@ -107,46 +138,82 @@ export function CardList({
-    if (cardsToShow.length === 0)
-      throw new Error(
-        "Sorry, we do not have any content in that category. But please check back later. We are constantly adding new content!"
-      );
-    setCards(cardsToShow.sort(() => Math.random() - 0.5));
-    setCurrentIndex(cardsToShow.length - 1);
+    setup();
   }, []);
 
   return (
     <Container fluid>
-      <Row className="d-grid justify-content-center" role="list">
-        {cards.map((card, index) => (
-          <CardItem
-            apiRef={childRefs[index]}
-            key={card.link}
-            data={card}
-            handleSwipe={(dir: string) => swiped(dir, index)}
-          />
-        ))}
-      </Row>
-      <Row className="mt-3 justify-content-between">
-        <Col md={3} xs={1} />
-        <Col className="d-flex align-items-center justify-content-center">
-          <Button onClick={() => swipe("left")}>{"<"} Left to skip!</Button>
-        </Col>
-        {/* <Col>
-          <Button onClick={() => goBack()}>Undo swipe!</Button>
-        </Col> */}
-        <Col className="d-flex align-items-center justify-content-center">
-          <Button onClick={() => swipe("right")} variant="secondary">
-            Right for a quiz! {">"}
-          </Button>
-        </Col>
-        <Col md={3} xs={1} />
-      </Row>
+      {(currentIndex > -1 && (
+        <>
+          <Row className="d-grid justify-content-center" role="list">
+            {cards.map((card, index) => (
+              <CardItem
+                apiRef={childRefs[index]}
+                key={card.link}
+                data={card}
+                handleSwipe={(dir: string) => swiped(dir, index)}
+              />
+            ))}
+          </Row>
+          <Row className="mt-3 justify-content-between">
+            <Col md={3} xs={1} />
+            <Col className="d-flex align-items-center justify-content-center">
+              <Button aria-label="left-button" onClick={() => swipe("left")}>
+                {"<"} Left to skip!
+              </Button>
+            </Col>
+            {/* <Col>
+      <Button onClick={() => goBack()}>Undo swipe!</Button>
+    </Col> */}
+            <Col className="d-flex align-items-center justify-content-center">
+              <Button
+                onClick={() => swipe("right")}
+                variant="secondary"
+                aria-label="right-button"
+              >
+                Right for a quiz! {">"}
+              </Button>
+            </Col>
+            <Col md={3} xs={1} />
+          </Row>
+        </>
+      )) || (
+        <Row>
+          <Col md={3} xs={1} />
+          <Col>
+            <p>
+              Looks like you&apos;ve reached the end. Please check back later.
+              We add new content regularly!
+            </p>
+            {areSkippedOrMissedCards() && (
+              <p>
+                Click{" "}
+                <a
```

Comment:
Is there a reason to not use `Link` from `react-router-dom` here?

Reply:
Is bliss!

Reply:
I changed one to `Link` and it broke the tests. I do not know why that is, so leaving it as `a` for now.